lib/spack/spack/test/packaging.py

```diff
@@ -0,0 +1,236 @@
+    bindist.relocate_package(spec)
+    spack.store.db.reindex(spack.store.layout)
+
+    args = parser.parse_args(['install', '-f', '-y', str(spec)])
+    buildcache.install_tarball(spec, args)
+    buildcache.buildcache(parser, args)
+
+    args = parser.parse_args(['list'])
+    buildcache.buildcache(parser, args)
+
+    args = parser.parse_args(['list', 'trivial'])
+    buildcache.buildcache(parser, args)
+
+    args = parser.parse_args(['keys'])
+    buildcache.buildcache(parser, args)
+
+    relocate.needs_binary_relocation('relocatable')
+
+    relocate.macho_make_paths_rel('/Users/Shares/spack/pkgC/lib/libC.dylib',
+                                  '/Users/Shared/spack',
+                                  ('/Users/Shared/spack/pkgA/lib',
+                                   '/Users/Shared/spack/pkgB/lib',
+                                   '/usr/local/lib'),
+                                  ('/Users/Shared/spack/pkgA/libA.dylib',
+                                   '/Users/Shared/spack/pkgB/libB.dylib',
+                                   '/usr/local/lib/libloco.dylib'),
+                                  '/Users/Shared/spack/pkgC/lib/libC.dylib')
+
+    relocate.macho_make_paths_rel('/Users/Shared/spack/pkgC/bin/exeC',
+                                  '/Users/Shared/spack',
+                                  ('/Users/Shared/spack/pkgA/lib',
+                                   '/Users/Shared/spack/pkgB/lib',
+                                   '/usr/local/lib'),
+                                  ('/Users/Shared/spack/pkgA/libA.dylib',
+                                   '/Users/Shared/spack/pkgB/libB.dylib',
+                                   '/usr/local/lib/libloco.dylib'),
+                                  None)
+
+    relocate.macho_replace_paths('/Users/Shared/spack',
+                                 '/Applications/spack',
+                                 ('/Users/Shared/spack/pkgA/lib',
+                                  '/Users/Shared/spack/pkgB/lib',
+                                  '/usr/local/lib'),
+                                 ('/Users/Shared/spack/pkgA/libA.dylib',
+                                  '/Users/Shared/spack/pkgB/libB.dylib',
+                                  '/usr/local/lib/libloco.dylib'),
+                                 '/Users/Shared/spack/pkgC/lib/libC.dylib')
+
+    relocate.macho_replace_paths('/Users/Shared/spack',
+                                 '/Applications/spack',
+                                 ('/Users/Shared/spack/pkgA/lib',
+                                  '/Users/Shared/spack/pkgB/lib',
+                                  '/usr/local/lib'),
+                                 ('/Users/Shared/spack/pkgA/libA.dylib',
+                                  '/Users/Shared/spack/pkgB/libB.dylib',
+                                  '/usr/local/lib/libloco.dylib'),
+                                 None)
+
+    if platform.system() == 'Darwin':
```

Comment:
I think this and the Linux platform check can be split out into separate unit tests.